Package: openoffice.org
Severity: normal

About the problem of the HelpLinker crash: g++-4.3 seems to miscompile
HelpLinker with -O2. Just changeing that to -O0 for the two directories
transex3/source/{,help/} solved that problem for us.

--- openoffice.org-3.2.0.o/ooo-build-3-2-0-6/patches/dev300/apply  2010-01-25 14:48:36.000000000 +0100
+++ openoffice.org-3.2.0/ooo-build-3-2-0-6/patches/dev300/apply  2010-01-25 15:07:23.000000000 +0100
@@ -293,6 +293,9 @@
 # it helps to avoid some crashes
 jvmfwk-sunjavaplugin-default-stack-size.diff, n#572372, pmladek
 
+# don't let HelpLinker be miscompiled with g++ 4.3
+gcc-HelpLinker.diff
+
 [ TemporaryHacks and not MacOSXOnly ]
 
 # don't lstat() that much while creating absolute URLs - breaks unopkg
--- openoffice.org-3.2.0.o/ooo-build-3-2-0-6/patches/dev300/gcc-HelpLinker.diff 1970-01-01 01:00:00.000000000 +0100
+++ openoffice.org-3.2.0/ooo-build-3-2-0-6/patches/dev300/gcc-HelpLinker.diff 2010-02-19 16:18:32.000000000 +0200
@@ -0,0 +1,20 @@
+--- transex3/source/makefile.mk.orig  2010-02-19 12:20:48.000000000 +0100
++++ transex3/source/makefile.mk 2010-02-18 19:49:58.000000000 +0100
+@@ -43,6 +43,7 @@
+ .INCLUDE :  settings.mk
+ CDEFS+= -DYY_NEVER_INTERACTIVE=1
+ #CDEFS+= -pg
++CDEFS+= -O0
+ 
+ .IF "$(SYSTEM_EXPAT)" == "YES"
+ CFLAGS+=-DSYSTEM_EXPAT
+--- transex3/source/help/makefile.mk.orig 2010-02-19 12:20:52.000000000 +0100
++++ transex3/source/help/makefile.mk  2010-02-19 11:45:26.000000000 +0100
+@@ -40,6 +40,7 @@
+ 
+ .INCLUDE : settings.mk
+ .INCLUDE : helplinker.pmk
++CDEFS+= -O0
+  
+ .IF "$(SYSTEM_LIBXSLT)" == "YES"
+ CFLAGS+= $(LIBXSLT_CFLAGS)
